ganga/GangaCore/Core/MonitoringComponent/MonitoringService.py

```diff
@@ -160,15 +160,16 @@ def _cleanup_dirty_jobs(self):
         submitted status.
         """
         for job_id in self.registry_slice.ids():
-            j = stripProxy(stripProxy(self.registry_slice(job_id)))
+            j = stripProxy(self.registry_slice(job_id))
             status = lazyLoadJobStatus(j)
-            if not j.subjobs and status == 'completing':
+            subjobs = lazyLoadJobObject(j, 'subjobs')
+            if not subjobs and status == 'completing':
                 j.status = 'running'
-            if j.subjobs:
-                if j.status == 'completing':
+            if subjobs:
+                if status == 'completing':
                     j.status = 'running'
                 for sj in j.subjobs:
```

Comment:
I think you need this to be indented, otherwise it loads any job with subjobs at line 171

Reply:
We do have to iterate through every job with subjobs in the current registry slice in order to make sure we fix all dirty jobs though. Consider the example where a job has multiple subjobs, and some of them were in the 'running' state while some others were in the 'completing' state when Ganga last exited. In this scenario, the master job status will be 'running' but some of the subjobs will have been stuck in the 'completing' state and will need to be reverted. So filtering by 'completing' master status will not pick those up.

I am just not sure why the caching issue happens when iterating through all subjobs. Shouldn't the monitoring component's registry slice be not stale when loaded? I guess a way to fix that would be to only select jobs in active master states when checking for dirty subjobs.

Reply:
By the way, is there a way to use the registry slice's `select` method to filter by multiple job states?